Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/ParentsOperation.php

```diff
@@ -65,26 +69,28 @@ public function canEvaluate($context)
      */
     public function evaluate(FlowQuery $flowQuery, array $arguments)
     {
-        $parents = [];
+        $parents = Nodes::createEmpty();
+        $findAncestorNodesFilter = FindAncestorNodesFilter::create(
+            NodeTypeConstraints::create(
+                NodeTypeNames::createEmpty(),
+                NodeTypeNames::fromArray(
+                    [NodeTypeName::fromString('Neos.ContentRepository:Root')]
+                )
+            )
```

Comment:
You could simplify this to:
```suggestion
            NodeTypeConstraints::fromFilterString('!Neos.ContentRepository:Root')
```

Not directly related to this change, but we should add some simpler factory methods to the `NodeTypeConstraints` VO – this is too much internal coupling IMO and it should be as simple as: `NodeTypeConstraints::withExplicitlyDisallowedNodeTypeNames('Neos.ContentRepository:Root', '...', '...')`.


Also: We should probably try to avoid magic strings and use the `NodeTypeNameFactory` instead

Reply:
Thought about that but wanted to avid the string parsing.

Reply:
That makes sense.
We can add better factory methods in a separate PR